lmnet/lmnet/networks/classification/vgg7.py

```diff
@@ -0,0 +1,268 @@
+
+        self.conv5 = self.conv_layer("conv5", self.pool2, filters=512, kernel_size=3, is_training=is_training)
+        self.conv6 = self.conv_layer("conv6", self.conv5, filters=512, kernel_size=3, is_training=is_training)
+
+        self.pool3 = self.max_pool("pool3", self.conv6, kernel_size=2, strides=2)
+
+        self.flatten = tf.contrib.layers.flatten(self.pool3)
+
+        self.fc1 = self.fc_layer("fc1", self.flatten, filters=1024, activation=tf.nn.relu)
+        self.fc1_drop = tf.nn.dropout(self.fc1, keep_prob)
+        self.fc2 = self.fc_layer("fc2", self.fc1_drop, filters=self.num_classes, activation=None)
+
+        return self.fc2
+
+    def conv_layer(
+        self,
+        name,
+        inputs,
+        filters,
+        kernel_size,
+        is_training=True,
+        strides=1,
+        padding="SAME",
+        activation=tf.nn.relu,
+        *args,
+        **kwargs
+    ):
+        kernel_initializer = tf.contrib.layers.xavier_initializer()
+        biases_initializer = tf.zeros_initializer()
+
+        if self.data_format == 'NHWC':
+            data_format = 'channels_last'
+        else:
+            data_format = 'channels_first'
+
+        #with tf.variable_scope(name, custom_getter=self.custom_getter):
+        conv = tf.layers.conv2d(inputs=inputs,
+                                filters=filters,
+                                kernel_size=kernel_size,
+                                kernel_initializer=kernel_initializer,
+                                #activation=None,
+                                #bias_initializer=biases_initializer,
+                                use_bias=False,
+                                padding=padding,
+                                strides=strides,
+                                data_format=data_format,
+                                kernel_regularizer=tf.contrib.layers.l2_regularizer(scale=0.0001),
+        )
+
+        batch_normed = tf.contrib.layers.batch_norm(conv,
+                                                    decay=0.99,
+                                                    scale=True,
+                                                    center=True,
+                                                    updates_collections=None,
+                                                    is_training=is_training,
+                                                    data_format=self.data_format)
+
+        #bias = tf.get_variable('bias', shape=filters, initializer=tf.zeros_initializer)
+        #biased = batch_normed + bias
+
+        output = tf.nn.relu(batch_normed)
+
+        return output
+
+    def fc_layer(
+            self,
+            name,
+            inputs,
+            filters,
+            activation,
+            *args,
+            **kwargs
+    ):
+        kernel_initializer = tf.contrib.layers.xavier_initializer()
+        biases_initializer = tf.zeros_initializer()
+
+        output = tf.contrib.layers.fully_connected(
```

Comment:
It looks not be applied weight decay.

Reply:
From this code, I think that the weight decay is automatically applied if self.weight_decay_rate is set.
Did I misunderstand?

https://github.com/blue-oil/blueoil/blob/2a04fccb507f6ec7f88337162eee3aa3669e14fc/lmnet/lmnet/networks/classification/base.py#L113

Reply:
Yes you misunderstand.

`weight_decay_loss` come from here
https://github.com/blue-oil/blueoil/blob/57d39650988d83684b72897ed2d9ac4cca6e1990/lmnet/lmnet/networks/classification/base.py#L85

Only the operation name contain `kernel` is applied `weight_decay_rate`.
fc layer's weight name is no `kernel`

Reply:
I got it. 
I will add weights_regularizer into the fully_connected layers.